src/cloudwatch-appsignals-mcp-server/evals/core/metrics_tracker.py

```diff
@@ -0,0 +1,189 @@
+        return len({c['tool_name'] for c in self.tool_calls})
+
+    @property
+    def task_duration(self) -> float:
+        """Calculate task duration in seconds."""
+        if self.task_start_time and self.task_end_time:
+            return self.task_end_time - self.task_start_time
+        return 0.0
+
+    @property
+    def tool_breakdown(self) -> Dict[str, Dict[str, int]]:
+        """Calculate per-tool call statistics."""
+        breakdown = {}
+        for call in self.tool_calls:
+            tool_name = call['tool_name']
+            if tool_name not in breakdown:
+                breakdown[tool_name] = {'count': 0, 'success': 0, 'failed': 0}
+            breakdown[tool_name]['count'] += 1
+            if call['success']:
+                breakdown[tool_name]['success'] += 1
+            else:
+                breakdown[tool_name]['failed'] += 1
+        return breakdown
+
+    @property
+    def file_operation_count(self) -> int:
+        """Return count of file operation tool calls."""
+        return len(self._get_file_operation_calls())
+
+    @property
+    def file_read_count(self) -> int:
+        """Return count of read_file calls."""
+        return len(
+            [c for c in self._get_file_operation_calls() if c['tool_name'] == FILE_TOOL_READ_FILE]
+        )
+
+    @property
+    def file_write_count(self) -> int:
+        """Return count of write_file calls."""
+        return len(
+            [c for c in self._get_file_operation_calls() if c['tool_name'] == FILE_TOOL_WRITE_FILE]
+        )
+
+    def _get_file_operation_calls(self) -> List[Dict[str, Any]]:
+        """Get all file operation tool calls."""
+        return [
+            c
+            for c in self.tool_calls
+            if c['tool_name'] in [FILE_TOOL_LIST_FILES, FILE_TOOL_READ_FILE, FILE_TOOL_WRITE_FILE]
+        ]
+
+    def compare_expected_tools(self, expected_tools: List[str]) -> Dict[str, Any]:
+        """Compare called tools against expected tools.
+
+        Args:
+            expected_tools: List of MCP tools expected to be used
+
+        Returns:
+            Dictionary with hit_rate, expected_tools_called, missing_expected_tools, unexpected_tools_called
+        """
+        expected_tool_set = set(expected_tools)
+        called_tool_names = {c['tool_name'] for c in self.tool_calls}
+        called_expected = called_tool_names & expected_tool_set
+        missing = expected_tool_set - called_tool_names
+        unexpected = called_tool_names - expected_tool_set
+
+        return {
+            'hit_rate': len(called_expected) / len(expected_tool_set)
+            if expected_tool_set
+            else 0.0,
+            'expected_tools_called': sorted(called_expected),
+            'missing_expected_tools': sorted(missing),
+            'unexpected_tools_called': sorted(unexpected),
+        }
+
+    def get_metrics(self, expected_tools: Optional[List[str]] = None) -> Dict[str, Any]:
```

Comment:
```suggestion
    def get_metrics_report(self, expected_tools: Optional[List[str]] = None) -> Dict[str, Any]:
```

Since it's only for reporting, I think?

Reply:
I think so too - renamed to `get_metrics_report()`